### DISCUSSION TITLE
Order relation array?

### DISCUSSION THREAD

**Discussion by conduongtong** [prisma/prisma#22561] [Q&A] [answered]
### Question

For example I have this update:
```ts
Await prisma.post.update({ where: { id: 1 }, data: { tags: { connect: [{ id: 2 }, { id: 1 }, { id: 3 }] } } });
```

When I use:
```ts
await prisma.post.findFirst({ include: { tags: true } });
```

It will return: [1,2,3]. Event if I change database to 2,1,3, it still return 1,2,3.
How can I order by like [2,1,3] without having order field?
<img width="268" alt="Screenshot 2024-01-04 at 4 56 14 PM" src="https://github.com/prisma/prisma/assets/43648013/5ae2e728-0bf0-4f16-acc4-0e8c562f10a8">

### Expected behavior (optional)

Return [2,1,3]

**Comment by nurul3101** [comment]
Hello @conduongtong 👋 

I don't believe it's possible without using the order field. Which database are you using?
You can do something like this, but it would still involve using order by.

```ts
const orderedIds = [2, 1, 3]; // The desired order of IDs
const result = await prisma.$queryRaw`SELECT * FROM Post WHERE id IN (${Prisma.join(orderedIds)}) ORDER BY FIELD(id, ${Prisma.join(orderedIds)})`;
```

Reference for [$queryRaw](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries#queryraw).

**Reply by conduongtong** [reply]
Thanks for reply.
I hope there will be a feature that do not sort by default like this. I cannot create another table for all of my relationship like this.
btw I'm using Postgres

**Reply by janpio** [reply, marked as answer]
Related issue opened by @conduongtong: https://github.com/prisma/prisma/issues/23898
